python/cudf/dataframe/dataframe.py

```diff
@@ -461,18 +461,23 @@ def take(self, positions, ignore_index=False):
             out[col] = self[col].take(positions, ignore_index=ignore_index)
         return out
 
-    def copy(self):
+    def copy(self, deep=True):
         """
         Returns a copy of this dataframe
         """
         df = DataFrame()
         df._index = self._index
         df._size = self._size
-        df._cols = self._cols.copy()
+        if deep:
+            for k in self._cols:
+                df[k] = Series(self._cols[k])
+                df[k]._column = self._cols[k]._column.copy()
```

Comment:
Instead of creating a new Series using the class initialization, I think we should implement the copy command at the Series level, as a user might want to shallow / deep copy a Series directly. Then this would just be `df[k] = df[k].copy()`

Reply:
Will do, coming up shortly.